### ISSUE TITLE
Bug: data loss

### ISSUE THREAD

**Issue by SamHasler** [choovick/ps-jsonutils#11]
```
> Get-JsonDifference -FromJsonString '{"hat": {"rabbit": "Fluffy"}}' `
>>                     -ToJsonString '{"hat": {"magic words to make the rabbit vanish": "abbacadabra"}}'
{
  "Added": {
    "hat": {
      "magic words to make the rabbit vanish": "abbacadabra"
    }
  },
  "Changed": {},
  "ChangedOriginals": {},
  "Removed": {},
  "NotChanged": {},
  "New": {
    "hat": {
      "magic words to make the rabbit vanish": "abbacadabra"
    }
  }
}
```

**Comment by SamHasler** [comment]
Or if you want something even more succinct:
```
> Get-JsonDifference -FromJsonString '{"🎩": {"🐇": ""}}' `
>>                     -ToJsonString '{"🎩": {"🧙🐇": ""}}'
{
  "Added": {
    "🎩": {
      "🧙🐇": ""
    }
  },
  "Changed": {},
  "ChangedOriginals": {},
  "Removed": {},
  "NotChanged": {},
  "New": {
    "🎩": {
      "🧙🐇": ""
    }
  }
}
```

**Comment by SamHasler** [comment]
It could be argued that the above is just handling renaming the key, but then there's this:

```
> Get-JsonDifference -FromJsonString '{"🎩": {"🧙🐇": "", "🐇":""}}' `
>>                     -ToJsonString '{"🎩": {"🧙🐇": ""}}'
{
  "Added": {},
  "Changed": {},
  "ChangedOriginals": {},
  "Removed": {},
  "NotChanged": {
    "🎩": {
      "🧙🐇": ""
    }
  },
  "New": {}
}
```

The particular case where this was spotted was with {"title":"...", "subtitle":"..."} and {"subtitle":"..."}

**Comment by SamHasler** [comment]
I think in the 3 uses of ` -match ` you need to surround the match regex with `^` and `$` to only match when it matches the whole string.

**Comment by Szeraax** [comment]
Thank you SO MUCH for contributing to this project! I love it when people find new ways to break things. I've just completed a code review of your PR and would like your feedback on my comments.

**Comment by Szeraax** [comment]
Stupid buttons!

**Comment by SamHasler** [comment]
And thank you for this project. It made writing the script I wanted to write so much easier. (Machine translation of json files.)

**Comment by Szeraax** [comment]
Glad to hear you're getting use out of it! This issue is now fixed via [ffa2041](https://github.com/choovick/ps-jsonutils/commit/ffa204149ad9ac6a24fe545fb4b6c9b032126d75) and you have been mentioned in the commit message should anyone come along later trying to find out who solved that crazy issue :D

It is also published now and you can update-module to get the latest version: https://www.powershellgallery.com/packages/JsonUtils/0.4.1

**Comment by SamHasler** [comment]
Works great, thank you!
